dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -63,13 +71,68 @@ def __add__(self, other: CategoricalColumn) -> CategoricalColumn:
             )
 
         merged_profile = CategoricalColumn(None)
-        merged_profile._categories = utils.add_nested_dictionaries(
-            self._categories, other._categories
-        )
-        BaseColumnProfiler._add_helper(merged_profile, self, other)
-        self._merge_calculations(
-            merged_profile.__calculations, self.__calculations, other.__calculations
-        )
+
+        # If both profiles have not met stop condition
+        if not (self._stop_condition_is_met or other._stop_condition_is_met):
+            merged_profile._categories = utils.add_nested_dictionaries(
+                self._categories, other._categories
+            )
+
+            BaseColumnProfiler._add_helper(merged_profile, self, other)
+            self._merge_calculations(
+                merged_profile.__calculations, self.__calculations, other.__calculations
+            )
+
+            # Transfer stop condition variables of 1st profile object to merged profile
+            # if they are not None else set to 2nd profile
+            if self.max_sample_size_to_check_stop_condition is not None:
+                merged_profile.max_sample_size_to_check_stop_condition = (
+                    self.max_sample_size_to_check_stop_condition
+                )
+            else:
+                merged_profile.max_sample_size_to_check_stop_condition = (
+                    other.max_sample_size_to_check_stop_condition
+                )
+            if self.stop_condition_unique_value_ratio is not None:
+                merged_profile.stop_condition_unique_value_ratio = (
+                    self.stop_condition_unique_value_ratio
+                )
+            else:
+                merged_profile.stop_condition_unique_value_ratio = (
+                    other.stop_condition_unique_value_ratio
+                )
+
+            # 1st and 2nd profile values should be same so no need for conditonal set
+            merged_profile._stop_condition_is_met = self._stop_condition_is_met
+            merged_profile._stopped_at_unique_ratio = self._stopped_at_unique_ratio
+            merged_profile._stopped_at_unique_count = self._stopped_at_unique_count
```

Comment:
as above, default should suffice in this case.

Reply:
same comment as above